samples/algorithms/HiddenShiftNISQ.qs

```diff
@@ -0,0 +1,168 @@
+/// # Sample
+/// Hidden shift
+///
+/// # Description
+/// There is a family of problems known as hidden shift problems, in which it
+/// is given that two Boolean functions 𝑓 and 𝑔 satisfy the relation
+///     𝑔(𝑥) = 𝑓(𝑥 ⊕ 𝑠) for all 𝑥
+/// where 𝑠 is a hidden bit string that we would like to find.
+///
+/// This Q# program implements an algorithm to solve the hidden shift problem.
+namespace Sample {
+    open Microsoft.Quantum.Arithmetic;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Measurement;
+
+    @EntryPoint()
+    operation Main() : Result[] {
+        // Consider the case of finding a hidden shift 𝑠 between two Boolean
+        // functions 𝑓(𝑥) and 𝑔(𝑥) = 𝑓(𝑥 ⊕ 𝑠).
+        // This problem can be solved on a quantum computer with one call to
+        // each of 𝑓 and 𝑔 in the special case that both functions are bent;
+        // that is, that they are as far from linear as possible.
+
+        // Here, we find the hidden shift encoded in the following array of
+        // Booleans.
+        let shiftAsBoolArray = [true, false, false, false, false, true];
+        let shiftAsInt = BoolArrayAsInt(shiftAsBoolArray);
+        let hiddenShiftBitString = FindHiddenShift(
+            BentFunction,
+            register => ShiftedBentFunction(shiftAsInt, register),
+            Length(shiftAsBoolArray));
+
+        return hiddenShiftBitString;
+    }
+
+    /// # Summary
+    /// Implements a correlation-based algorithm to solve the hidden shift
+    /// problem for bent functions.
+    ///
+    /// # Description
+    /// Implements a solution for the hidden shift problem, which is to identify
+    /// an unknown shift 𝑠 of the arguments of two Boolean functions 𝑓 and 𝑔
+    /// that are promised to satisfy the relation 𝑔(𝑥) = 𝑓(𝑥 ⊕ 𝑠) for all 𝑥.
+    ///
+    /// 𝑓 and 𝑔 are assumed to be bent functions. A Boolean function is bent if
+    /// it is as far from linear as possible. In particular, bent functions have
+    /// flat Fourier (Walsh–Hadamard) spectra.
+    ///
+    /// In this case, the Roetteler algorithm (see References, below) uses
+    /// black-box oracles for 𝑓^* and 𝑔, where 𝑓^* is the dual bent function to
+    /// 𝑓, and computes the hidden shift 𝑠 between 𝑓 and 𝑔.
+    ///
+    /// # Input
+    /// ## Ufstar
+    /// A quantum operation that implements
+    /// $U_f^*: |𝑥〉 ↦ (-1)^{f^*(x)} |𝑥〉$,
```

Comment:
Should this be converted to Unicode?

Reply:
I'm not sure this can completely be expressed in unicode.